flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/sink/throwable/FatalExceptionClassifier.java

```diff
@@ -44,13 +44,13 @@ public FatalExceptionClassifier(
     public boolean isFatal(Throwable err, Consumer<Exception> throwableConsumer) {
         if (validator.test(err)) {
             throwableConsumer.accept(throwableMapper.apply(err));
-            return false;
+            return true;
```

Comment:
@DavidLiu001 thanks for the PR but this change is very dangerous as-is, since we are inverting the semantics of the code. This is not a backward compatible change. Instead of changing this, can we 1/ `@Deprecate` the existing implementation and 2/ introduce new code that is cleaner. We can remove the deprecated version in a future Flink version.

Side comment I am concerned that no tests failed with this! Seems there is a gap in the unit tests. Can you please add a test for this or raise a Jira to cover that?

Reply:
@dannycranmer

Reply:
> Can you please add a test for this or raise a Jira to cover that?

I would say that a test must be added before a PR like this can be merged. We shouldn't do this later. 

@dannycranmer If this is `@Internal` I think we can break it between versions, unless there are things that rely on this and it actually should have been made public. WDYT?